### ISSUE TITLE
Regexp in UTF-8 causes issues in Lint/OutOfRangeRegexRef

### ISSUE THREAD

**Issue by m11o** [rubocop/rubocop#10552]
## Expected behavior

Lint/OutOfRangeRegexRef would not alert when using utf-8 encoding.

## Actual behavior

Rubocop alerted when using encoding utf-8 ( `/u` ). But, Rubocop didn't alert when not using it.

## Steps to reproduce the problem

Given the code:

```ruby
/^(\d+)-(\S+)$/u =~ value
if $1 != target
  # do something
end
```
=> NG

```
W: Lint/OutOfRangeRegexpRef: $1 is out of range (no regexp capture groups detected).
if $1 != target
   ^^
```

```ruby
/^(\d+)-(\S+)$/ =~ value
if $1 != target
  # do something
end
```
=> OK

## RuboCop version

```
$ bundle exec rubocop -V
1.25.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 2.7.5 x86_64-linux)
  - rubocop-rails 2.13.2
```

Please tell me why they occur. Thank you!

**Comment by m11o** [comment]
Appendix

```ruby
/^(\d+)-(\S+)$/u =~ value
if Regexp.last_match(1) != target
  # do something
end
```
=> OK

**Comment by marcandre** [comment]
Looks good :-)

**Comment by koic** [comment]
@marcandre I've opened https://github.com/rubocop/rubocop-ast/pull/230 to resolve this issue. Can you review it?
